File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.java
class Solution {
    public int subarraySum(int[] arr, int k) {
        int n = arr.length;
        int cs = 0;
        int count = 0;

        HashMap<Integer,Integer> map = new HashMap<>();
        map.put(0,1);
        for(int i : arr){
            cs +=i;
            int diff = cs-k;

            if(map.containsKey(diff)){
                count+=map.get(diff);
            }

            map.put(cs,map.getOrDefault(cs,0)+1);

        }
        return count;
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, arr: List[int]) -> int:
        count = 0
        maj = arr[0]
        for i in range(len(arr)):
            if count==0:
                maj = arr[i]
                count+=1
            elif arr[i]==maj:
                count+=1
            else:
                count-=1
        
        return maj